### PR TITLE
Replace phone number with coupon code

### PR DESCRIPTION
**Pull request by nelsontky**
Some of the test cases still have fields that look like phone numbers

### CODE REVIEW

File: src/main/java/csdev/couponstash/logic/parser/AddCommandParser.java [hunk 1]
```diff
@@ -45,20 +46,22 @@ public AddCommandParser(String moneySymbol) {
...
-        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_PHONE, PREFIX_EXPIRY_DATE)
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_EXPIRY_DATE)
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
         }
 
         Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
-        Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
+        PromoCode promoCode = ParserUtil.parsePromoCode(
+                argMultimap.getValueForOptionalField(PREFIX_PROMO_CODE, "-").get());
```
Comment (ChesterSim) [inline src/main/java/csdev/couponstash/logic/parser/AddCommandParser.java:64]: Good use of getValueForOptionalField()!

Comment (nelsontky) [reply src/main/java/csdev/couponstash/logic/parser/AddCommandParser.java:64]: Lol didn't expect you to actually code review upz la good job!

File: src/main/java/csdev/couponstash/model/coupon/PromoCode.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package csdev.couponstash.model.coupon;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a Coupon's promo code in the CouponStash.
+ * Guarantees: immutable
+ */
+public class PromoCode {
+
```
Comment (ChesterSim) [inline src/main/java/csdev/couponstash/model/coupon/PromoCode.java:10]: Good consideration to not limit the promo code to any validation regex!

Comment (nelsontky) [reply src/main/java/csdev/couponstash/model/coupon/PromoCode.java:10]: That's cuz I dunno regex :)

**Review by kevvvinn — APPROVED** [review]
Nicely written

**Pull request merged**